tiledb/common/interval/test/unit_interval_bound.cc

```diff
@@ -0,0 +1,321 @@
+/**
+ * @file unit_interval_bound.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2021 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file defines tests for the Interval::Bounds inner class.
+ *
+ * The comparison tests for bounds do not include the null bound in the tests.
+ * The simplest reason to state is that they're excluded by precondition from
+ * the comparison functions, so we can't include them. More broadly, though,
+ * empty bounds aren't considered ordered, and the empty set is always treated
+ * separately in any of the comparison functions.
+ */
+
+#include "unit_interval.h"
+
+TEMPLATE_LIST_TEST_CASE(
```

Comment:
@ihnorton There is some pretty nice catch2 usage to dynamically generate the tests.